Examples/Io/Root/src/RootVertexPerformanceWriter.cpp

```diff
@@ -0,0 +1,280 @@
+    const std::vector<Acts::Vertex<Acts::BoundTrackParameters>>& vertices) {
+  // Exclusive access to the tree while writing
+  std::lock_guard<std::mutex> lock(m_writeMutex);
+
+  m_nrecoVtx = vertices.size();
+
+  ACTS_DEBUG("Number of reco vertices in event: " << m_nrecoVtx);
+  if (m_outputFile == nullptr)
+    return ProcessCode::SUCCESS;
+
+  // Read truth particle input collection
+  const auto& allTruthParticles =
+      ctx.eventStore.get<SimParticleContainer>(m_cfg.inputAllTruthParticles);
+  // Get number of true primary vertices
+  m_ntrueVtx = getNumberOfTruePriVertices(allTruthParticles);
+
+  ACTS_INFO("Total number of generated truth particles in event : "
+            << allTruthParticles.size());
+  ACTS_INFO(
+      "Total number of generated truth primary vertices : " << m_ntrueVtx);
+
+  // Read selected truth particle input collection
+  const auto& allSelectedParticles = ctx.eventStore.get<SimParticleContainer>(
+      m_cfg.inputSelectedTruthParticles);
+  // Get number of true primary vertices
+  m_nVtxDetAcceptance = getNumberOfTruePriVertices(allSelectedParticles);
+
+  ACTS_INFO("Total number of selected truth particles in event : "
+            << allSelectedParticles.size());
+  ACTS_INFO("Total number of detector-accepted truth primary vertices : "
+            << m_nVtxDetAcceptance);
+
+  // Read track-associated truth particle input collection
+  const auto& allAssociatedTruthParticles =
+      ctx.eventStore.get<SimParticleContainer>(
+          m_cfg.inputAssociatedTruthParticles);
+  // Get number of true primary vertices
+  m_nVtxReconstructable =
+      getNumberOfReconstructableVertices(allAssociatedTruthParticles);
+
+  ACTS_INFO("Total number of reco track-associated truth particles in event : "
+            << allAssociatedTruthParticles.size());
+  ACTS_INFO("Total number of reco track-associated truth primary vertices : "
+            << m_nVtxReconstructable);
+
+  /*****************  Start x,y,z resolution plots here *****************/
+  // Matching tracks at vertex to fitted tracks that are in turn matched
+  // to truth particles. Match reco and true vtx if >50% of tracks match
+
+  const auto& allFittedTracks =
+      ctx.eventStore.get<std::vector<Acts::BoundTrackParameters>>(
+          m_cfg.allFittedTracks);
+
+  // Loop over all reco vertices and find associated truth particles
+  std::vector<SimParticleContainer> truthParticlesAtVtxContainer;
+  for (const auto& vtx : vertices) {
+    const auto tracks = vtx.tracks();
+    // Store all associated truth particles to current vtx
+    SimParticleContainer particleAtVtx;
+
+    std::vector<int> contributingTruthVertices;
+
+    for (const auto& trk : tracks) {
+      Acts::BoundTrackParameters origTrack = *(trk.originalParams);
+
+      // Find associated truth particle now
+      int idx = 0;
+      for (const auto& particle : allAssociatedTruthParticles) {
+        if (origTrack.parameters() == allFittedTracks[idx].parameters()) {
```

Comment:
So the assumption is that, the `allAssociatedTruthParticles` and the `allFittedTracks` have one-to-one mapping? Is it always true?

Reply:
Yes, exactly. This should be always true. However, I added a small check to be safe here.